File: year2/semester1/java/src/main/java/domain/Grade.java
package domain;

import java.time.LocalDate;

public class Grade extends Entity<GradeID> {
    private LocalDate date;
    private String professor;
    private float grade;

    public Grade(float grade, LocalDate date, String professor, String idStudent, String idAssignment) {
        this.grade = grade;
        this.date = date;
        this.professor = professor;
        setId(new GradeID(idStudent, idAssignment));
    }

    public LocalDate getDate() {
        return date;
    }

    public String getProfessor() {
        return professor;
    }

    public float getGrade() {
        return grade;
    }

    public void setGrade(float grade) {
        this.grade = grade;
    }

    @Override
    public String toString() {
        return "student ID: " + getId().getStudentId() + " assignment ID: " + getId().getAssignmentId() + " " + grade;
    }
}

File: year1/semester1/fp/domain/grade.py
class Grade:

    def __init__(self, student, course, grade):
        self.__student = student
        self.__course = course
        self._grade = grade

    def get_student(self):
        return self.__student

    def get_course(self):
        return self.__course

    def get_grade(self):
        return self._grade

    def set_grade(self, new_grade):
        self._grade = new_grade

    def __str__(self):
        return "Student: " + str(self.__student) + " Course: " + str(self.__course) + " Grade: " + str(self._grade)


